single_tech_samples/fabric/feature_engineering_on_fabric/README.md

```diff
@@ -0,0 +1,373 @@
+  - __client_secret__: Type: String; Default Value: `@pipeline().parameters.client_secret`
+
+- Step 4, Within the 'Settings' tab, select our current workspace and then choose the right Notebook file.
+  - __Data Ingestion__ -> __data_ingestion__
+  - __Data Cleansing__ -> __data_cleansing__
+  - __Data Transformation__ -> __data_transformation__
+  - __Feature Registration__ -> __feature_set_registration__
+  
+  ![data_pipeline_15](./images/data_pipeline/data_pipeline_15.png)
+
+### Feature Store Setup
+
+Follow the official documentation to create a feature store:
+[Create a Minimal Feature Store](https://learn.microsoft.com/en-us/azure/machine-learning/tutorial-get-started-with-feature-store?view=azureml-api-2&tabs=SDK-track#create-a-minimal-feature-store).
+
+Create a Service Principle and make a memo of the client secret, client id and tenant id.
+![service principle](./images/featurestore_2.png)
+
+Go to the page of feature store in Azure portal, and assign the Service Principle with following role:
+
+- `AzureML Data Scientist`
+
+  ![roles](./images/featurestore_1.png)
+
+### Fabric Environment Setup
+
+Go to Fabric workspace homepage, create a new __Environment__ by clicking __+ New__ button in workspace homepage and select __Environment__ under __Data Engineering__.
+![new](./images/featurestore_5.png)
+![env](./images/featurestore_4.png)
+
+In the Environment, you will need to setup the python env under Public Libraries. You can do this by adding `azureml-featurestore` by PyPI, or by clicking __Add from yml__ and select `./src/environment/Publiclibrary.yml` under this repo.
+
+![pip](./images/featurestore_3.png)
+
+You will also need to edit the Spark properties. Find the properties list yaml template in this repo `./src/environment/sparkProperties.yaml`. Replace the value with your created resources of Azure Machine Learning Managed Feature Store and Service Principle.
+
+```yaml
+runtime_version: '1.1'
+spark_conf:
+  - spark.fsd.client_id: <sp-client-id>
+  - spark.fsd.tenant_id: <sp-tenant-id>
+  - spark.fsd.subscription_id: <subscription-id>
+  - spark.fsd.rg_name: <feature-store-resouce-group>
+  - spark.fsd.name: <feature-store-name>
+  - spark.fsd.fabric.tenant: <fabric-tenant-name> # Fetch from Fabric base URL, like https://<fabric-tenant-name>.powerbi.com/
+  - spark.fsd.fabric.workspace: <fabric-workspace>
+  - spark.fsd.fabric.lakehouse: <fabric-lakehouse>
+  - spark.fsd.purview.account: <purview-account-name>
+
+```
+
+After this is done, click __Save__ to save the environment, and then __Publish__. This may take a few minutes to finish.
+
+To apply the environment, you can set the newly created environment as 'default' in the Fabric workspace settings page.
+
+![pip](./images/fabric_env_2.png)
+
+Or you can apply to each specific notebook in the notebook edition page.
+
+![pip](./images/fabric_env_1.png)
+
+### Model Training and Inferencing Setup
```

Comment:
This seems a bit out-of-place. Can we not add it later, when we start talking about the Feature registration.